### PR TITLE
E2e metric

### LINKED ISSUE #770: [E2E Enhancement] Add e2e tests covering L4 telemetry
**What would you like to be added**:
 e2e tests covering L4 telemetry
**Why is this needed**:
After adding telemetry-related functionality to kmesh, it is necessary to add a related e2e case to ensure that subsequent modifications do not damage telemetry functionality.

### PR DESCRIPTION
**Pull request by LiZhenCheng9527**
**What type of PR is this?**
/kind enhancement

**What this PR does / why we need it**:
add L4 Telemetry e2e case
**Which issue(s) this PR fixes**:
Fixes #770 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: test/e2e/main_test.go [hunk 1]
```diff
@@ -148,10 +165,10 @@ func SetupApps(t resource.Context, i istio.Instance, apps *EchoDeployments) erro
 				},
 				{
 					Replicas: 1,
-					Version:  "v2",
+					Version:  "v1",
```
Comment (YaoZengzeng) [inline test/e2e/main_test.go:168]: What was written before is correct. 

A subset represents a deployment corresponding to a service. A service can have multiple subsets representing deployments of different versions.

File: test/e2e/main_test.go [hunk 2]
```diff
@@ -112,6 +116,15 @@ func TestMain(m *testing.M) {
 		Setup(func(t resource.Context) error {
 			return SetupApps(t, i, apps)
 		}).
+		Setup(
+			func(t resource.Context) (err error) {
```
Comment (YaoZengzeng) [inline test/e2e/main_test.go:120]:
```suggestion
			Setup(func(t resource.Context) (err error) {
			      prom, err = prometheus.New(t, prometheus.Config{})
			      return
			}
```

The format should be consistent with other `Setup`

File: test/e2e/baseline_test.go [hunk 3]
```diff
@@ -691,3 +697,142 @@ func SetWaypoint(t framework.TestContext, ns string, name string, waypoint strin
...
+	query.Labels = labels
+
+	return query
+}
+
+func deployName(inst echo.Instance) string {
+	return inst.ServiceName() + "-" + inst.Config().Version
+}
+
+func PromDiff(t test.Failer, prom prometheus.Instance, cluster cluster.Cluster, query prometheus.Query) {
```
Comment (YaoZengzeng) [inline test/e2e/baseline_test.go:783]: Why not just use the `PromDiff` function in `istio.io/istio/tests/integration/telemetry` ?

File: test/e2e/baseline_test.go [hunk 4]
```diff
@@ -691,3 +697,142 @@ func SetWaypoint(t framework.TestContext, ns string, name string, waypoint strin
 		}
 	}
 }
+
+func TestL4Telemetry(t *testing.T) {
+	framework.NewTest(t).Run(func(tc framework.TestContext) {
+		for _, src := range apps.All {
+			for _, dst := range apps.All {
```
Comment (YaoZengzeng) [inline test/e2e/baseline_test.go:704]:
```suggestion
                  for _, src := range apps.EnrolledToKmesh {
			for _, dst := range apps.EnrolledToKmesh {
```

`apps.All` contains all test applications. Although they are all currently managed by Kmesh, we may include unmanaged applications in the future. So use `apps.EnrolledToKmesh`  is sufficient, which means the applications managed by Kmesh

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

Please move telemetry test out, can be done later

**Pull request merged**